ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Impl/Index/Cache.hs

```diff
@@ -581,19 +587,34 @@ getEpochInfo cacheEnv epoch = do
   Operations
 ------------------------------------------------------------------------------}
 
--- | Empties the cache and restarts the background expiration thread.
+-- | Stops the background expiration thread.
+--
+-- This operation is idempotent.
+close :: IOLike m => CacheEnv m hash h -> m ()
+close CacheEnv { bgThreadVar } = do
+    mbBgThread <- atomically $ tryTakeTMVar bgThreadVar
+    mapM_ cancelThread mbBgThread
+
+-- | Restarts the background expiration thread, drops all previously cached
+-- information, loads the given epoch.
 --
 -- PRECONDITION: the background thread expiring unused past epochs must have
 -- been terminated.
-reset
+restart
   :: IOLike m
   => CacheEnv m hash h
   -> EpochNo  -- ^ The new current epoch
   -> m ()
-reset cacheEnv@CacheEnv { hasFS, err, hashInfo, registry, cacheVar } epoch = do
+restart cacheEnv epoch = do
```

Comment:
Add a comment that this is used only in the tests (it doesn't leak past the ImmutableDB, because the only code path that triggers it is part of the `Internal`r ecord) and that we don't have to worry about race conditions involving `bgThreadVar`.